src/main/java/com/easytestit/generatexml/GenerateXML.java

```diff
@@ -1,145 +1,163 @@
-        File resultsFolder = new File(DefaultData.REPORT_RESULTS_FOLDER);
-        LOGGER.info(!resultsFolder.exists() && resultsFolder.mkdirs() ? "Direction ".concat(DefaultData.REPORT_RESULTS_FOLDER).concat(" was created") : "Direction ".concat(DefaultData.REPORT_RESULTS_FOLDER).concat(" was already created"));
+        String path;
+        if (configuration == null) {
+            path = ConfigDataProvider.DEFAULT_FOLDER;
+        } else {
+            path = ConfigDataProvider.REPORT_RESULTS_FOLDER;
+        }
+
+        File resultsFolder = new File(path);
+        LOGGER.log(Level.INFO, !resultsFolder.exists() && resultsFolder.mkdirs()
+                ? "Direction " + path + " created successfully"
+                : "Direction " + path + " was already created"
+        );
     }
 
     /**
      * A convenience method which read files with specific extension
      * and collects path to these files in a new String collection
-     * </p>
-     * @param reportFolder the folder where locate all files after launching tests
+     *
+     * @param source the folder where locate all files after launching tests
      * @return the collection of String with defined path to JSON files
      */
-    @NotNull
-    private Collection<String> getJsonFilesFrom(@NotNull File reportFolder) {
-        var extensions = Collections.singletonList(DefaultData.DEFAULT_FILE_EXTENSIONS);
-        Collection<String> out = new ArrayList<>();
-        File[] files = reportFolder.listFiles();
-        //might be null
-        if (files == null) {
-            LOGGER.info("Report folder ".concat(reportFolder.getName()).concat(" aren't contain any files!"));
-            return out;
-        }
-        //analyze files in folder and filter by specified extension and collect these to new collection
-        Arrays.stream(files).forEach(file -> {
-            if (file.isFile() && extensions.contains(file.getName().substring(file.getName().lastIndexOf('.') + 1))) {
-                out.add(String.valueOf(reportFolder)
-                        .concat(String.valueOf(File.separatorChar))
-                        .concat(file.getName()));
-            } else {
-                LOGGER.info("Report folder ".concat(reportFolder.getName())
-                        .concat(" doesn't contain files with JSON format!"));
+    private Collection<String> resolvePath(final File source) {
+        Collection<String> filePathList = new ArrayList<>();
+        if (source != null) {
+            File[] files = source.listFiles();
+
+            if (files == null) {
+                throw new GenerateXMLReportException("Provided source folder " + source.getName() + " doesn't contain any files!");
             }
-        });
 
-        return out;
-    }
+            filePathList = Arrays.stream(files)
+                    .filter(File::isFile)
+                    .filter(file -> file.getName().endsWith("json"))
+                    .map(file -> String.valueOf(source)
```

Comment:
What kind of value do you expect here `String.valueOf(source)` ?

Reply:
I changed it to a more comfortable read in code.